src/redux/reducers.js

```diff
@@ -0,0 +1,94 @@
+import { combineReducers } from 'redux';
+import { createSelector } from 'reselect';
+import Inmutable from 'seamless-immutable';
+
+import { actions } from './actions';
+
+const defaultState = Inmutable({
+  wBooks: {
+    filterType: '',
+    filter: '',
+    books: [],
+    filteredBooks: []
+  },
+  isFetching: false,
+  bookDetail: null
+});
+
+const search = (text, filter) => (
+  text.toLowerCase().indexOf(filter.toLowerCase()) > -1
+);
+
+const booksSelector = state => state.books;
+const filterTypeSelector = state => state.filterType;
+const filterSelector = state => state.filter;
+
+const filteredBooksSelector = createSelector(
+  booksSelector,
+  filterTypeSelector,
+  filterSelector,
+  (books, filterType, filter) => books.filter(book => search(book[filterType], filter))
```

Comment:
What we usually do is to use these selectors inside a `mapStateToProps`, since there we have full access to the store and we are not constrained by a particular reducer. Nonetheless, I think this approach suits well here (at least for the filter task; API usually does filtering).